### PR TITLE
Don't error on duplicate prototype property assignments

### LINKED ISSUE #7063: Duplicate identifier in buffer and send packages
Using the allowJs compiler option, if [buffer](https://github.com/feross/buffer) or [send](https://github.com/pillarjs/send) are dependencies, Duplicate identifier errors are reported:
```log
node_modules/buffer/index.js(94,5): error TS2300: Duplicate identifier 'length'.
node_modules/buffer/index.js(95,5): error TS2300: Duplicate identifier 'parent'.
node_modules/buffer/index.js(229,20): error TS2300: Duplicate identifier 'length'.
node_modules/buffer/index.js(230,20): error TS2300: Duplicate identifier 'parent'.
node_modules/send/index.js(203,22): error TS2300: Duplicate identifier 'root'.
node_modules/send/index.js(212,22): error TS2300: Duplicate identifier 'root'.
```

It seems similar to #6645, but here in addition to this member assignment, prototypes are in play.

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes #7063

### CODE REVIEW

File: src/compiler/binder.ts [hunk 1]
```diff
@@ -1465,7 +1465,7 @@ namespace ts {
             }
 
             // Declare the method/property
-            declareSymbol(funcSymbol.members, funcSymbol, leftSideOfAssignment, SymbolFlags.Property, SymbolFlags.PropertyExcludes);
+            declareSymbol(funcSymbol.members, funcSymbol, leftSideOfAssignment, SymbolFlags.Property, SymbolFlags.PropertyExcludes & ~SymbolFlags.Property);
```
Comment (DanielRosenwasser) [inline src/compiler/binder.ts]: Adapt the above comment:

```ts
// It's acceptable for multiple 'Foo.prototype' assignments of the same identifier to occur
```

**Comment by mhegazy** [comment]
:+1:

File: tests/cases/fourslash/jsDocClasses1.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+///<reference path="fourslash.ts" />
+
+// @allowNonTsExtensions: true
```
Comment (mhegazy) [inline tests/cases/fourslash/jsDocClasses1.ts]: do not think you needed the fourslash test here. a compiler baseline test would have been more appropriate.

**Comment by billti** [comment]
:+1:

**Comment by DanielRosenwasser** [comment]
Ping - can you address the comments and merge this in @RyanCavanaugh?

**Pull request merged**